llvm/unittests/Support/KnownBitsTest.cpp

```diff
@@ -320,6 +320,22 @@ TEST(KnownBitsTest, AbsDiffSpecialCase) {
   EXPECT_EQ(0b0000ul, Res.Zero.getZExtValue());
 }
 
+TEST(KnownBitsTest, ShrExactSpecialCase) {
```

Comment:
I don't understand what this special case is - why isn't it covered by the exhaustive testing below?

Reply:
This was just to show a case where exact provides info that non-exact doesn't. But ill change this to an IR test where the codegen actually varies @dtcxzyw's request.